sharing/src/test/scala/io/delta/sharing/spark/DeltaSharingDataSourceDeltaSuite.scala

```diff
@@ -0,0 +1,571 @@
+          test(s"${profileFile.getCanonicalPath}#share1.default.$sharedTableName")
+        }
+      }
+    }
+  }
+
+  test("DeltaSharingDataSource able to read data with join on the same table") {
+    withTempDir { tempDir =>
+      val deltaTableName = "base_delta_table"
+      withTable(deltaTableName) {
+        createSimpleTable(deltaTableName, enableCdf = false)
+        sql(s"""INSERT INTO $deltaTableName VALUES (1, "first"), (2, "first")""")
+        sql(s"""INSERT INTO $deltaTableName VALUES (1, "second"), (2, "second")""")
+        sql(s"""INSERT INTO $deltaTableName VALUES (1, "third"), (2, "third")""")
+
+        val sharedTableName = "shared_table"
+        prepareMockedClientAndFileSystemResult(deltaTableName, sharedTableName)
+        prepareMockedClientGetTableVersion(deltaTableName, sharedTableName)
+        prepareMockedClientAndFileSystemResult(
+          deltaTableName,
+          sharedTableName,
+          versionAsOf = Some(1L)
+        )
+
+        def testJoin(tablePath: String): Unit = {
+          // Query the same latest version
+          val deltaDfLatest = spark.read.format("delta").table(deltaTableName)
+          val deltaDfV1 = spark.read.format("delta").option("versionAsOf", 1).table(deltaTableName)
+          val sharingDfLatest =
+            spark.read.format("deltaSharing").option("responseFormat", "delta").load(tablePath)
+          val sharingDfV1 =
+            spark.read
+              .format("deltaSharing")
+              .option("responseFormat", "delta")
+              .option("versionAsOf", 1)
+              .load(tablePath)
+
+          var deltaDfJoined = deltaDfLatest.join(deltaDfLatest, "c1")
+          var sharingDfJoined = sharingDfLatest.join(sharingDfLatest, "c1")
+          // CheckAnswer ensures that delta sharing produces the same result as delta.
+          // The check on the size is used to double check that a valid dataframe is generated.
+          checkAnswer(deltaDfJoined, sharingDfJoined)
+          assert(sharingDfJoined.count() > 0)
+
+          // Query the same versionAsOf
+          deltaDfJoined = deltaDfV1.join(deltaDfV1, "c1")
+          sharingDfJoined = sharingDfV1.join(sharingDfV1, "c1")
+          checkAnswer(deltaDfJoined, sharingDfJoined)
+          assert(sharingDfJoined.count() > 0)
+
+          // Query with different versions
+          deltaDfJoined = deltaDfLatest.join(deltaDfV1, "c1")
+          sharingDfJoined = sharingDfLatest.join(sharingDfV1, "c1")
+          checkAnswer(deltaDfJoined, sharingDfJoined)
+          // Size is 6 because for each of the 6 rows in latest, there is 1 row with the same c1
+          // value in v1.
+          assert(sharingDfJoined.count() > 0)
+        }
+
+        withSQLConf(getDeltaSharingClassesSQLConf.toSeq: _*) {
+          val profileFile = prepareProfileFile(tempDir)
+          testJoin(s"${profileFile.getCanonicalPath}#share1.default.$sharedTableName")
+        }
+      }
+    }
+  }
+}
+
```

Comment:
are CDC reading tests coming next?

Reply:
also reading delta sharing tables with DVs?

Reply:
yes, cdc support and test will come next, and dv is yet the next step.